pyang/plugins/uml.py

```diff
@@ -591,7 +615,12 @@ def emit_module_header(self, module, fd):
     def emit_module_class(self, module, fd):
         fd.write('class \"%s\" as %s << (M, #33CCFF) module>> \n' %(self.full_display_path(module), self.full_path(module)))
 
-
+    def emit_uml_legend(self, module, fd):
+        fd.write('legend \n')
```

Comment:
Should a legend be written if there are no files/modules to list, i.e., only when self.ctx_grp_files is empty? This is currently the case if I don't specify either --uml-inline-groupings or --uml-skip-module, but specify --uml-add-legend, then a legend is displayed without any files listed.

Also if --uml-inline-groupings and --uml-skip-module are not specified wouldn't it be expected that if a legend is specified, then the modules of the groupings should be listed as when --uml-inline-groupings=true and --uml-skip-module!={}?